### DISCUSSION TITLE
Path not set from config.nu

### DISCUSSION THREAD

**Discussion by Guara92** [nushell/nushell#14690] [Q&A]
Hi,

I recently updated my config.nu (on Linux) as described in the [doc](https://www.nushell.sh/book/configuration.html#common-configuration-tasks-in-config-nu), but it seems that is not working as expected, here is a snippet of my config.nu:
```nushell
use std/util "path add"
$env.CARGO_HOME = ($env.HOME | path join ".cargo")
$env.CARGO_BIN = ($env.CARGO_HOME | path join "bin")
path add $env.CARGO_BIN
```
but the var is not applied at shell startup
```
> echo $env.PATH
╭───┬─────────────────╮
│ 0 │ /usr/local/bin  │
│ 1 │ /usr/local/sbin │
│ 2 │ /usr/bin        │
│ 3 │ /usr/sbin       │
╰───┴─────────────────╯
```
If I add the same var to path manually is added correctly:
```
> path add $env.CARGO_BIN
> echo $env.PATH
╭───┬───────────────────────╮
│ 0 │ /home/user/.cargo/bin │
│ 1 │ /usr/local/bin        │
│ 2 │ /usr/local/sbin       │
│ 3 │ /usr/bin              │
│ 4 │ /usr/sbin             │
╰───┴───────────────────────╯
```

I am missing something?
Thanks

**Comment by ghost** [comment]
hi.  
for me `use std/util; util path add '~/.cargo/bin'` just works,  
but **in my personal experience** the stdlib is buggy (`std/log` requires me to manually set 3 env-vars before use, etc) and thus i avoid using it. I wouldnt be suprised if it would just happen to be broken under some circumstances.

I checked [the code for it](https://github.com/nushell/nushell/blob/main/crates/nu-std/std/util/mod.nu) and i don't see how it could fail without outputting an error.

a few things, which might help with finding the problem:

1. what is the output of `version | transpose key value | to md --pretty`? if the version is less than `0.101.0` it might be a already fixed bug.  
2. does it work if you run it in an interactive terminal (instead of the config)?  
3. does `http get "https://raw.githubusercontent.com/nushell/nushell/refs/heads/main/crates/nu-std/std/util/mod.nu" | save -r tmp.nu; use ./tmp.nu *; path add "~/.cargo/bin"` work?
4. try to add a `print` between each line in the `tmp.nu`, reopen nu, retry `use ./tmp.nu *; path add "~/.cargo/bin"` and send us where it failed.

**Reply by Guara92** [reply]
Hi,

1. Yes I'm on `0.101.0`
2. Yes
3. It works

I also tried with the syntax `$env.path ++= ["~/.cargo/bin"]` and is not working from `config.nu`, it works instead from interactive shell.
I rolled back to my old config with path set in `env.nu` 
```nushell
$env.PATH = ($env.PATH | split row (char esep) | prepend [$'($env.HOME)/.cargo/bin', $'($env.HOME)/.local/bin', $'($env.HOME)/bin', '/usr/local/bin', '/usr/local/sbin', '/usr/bin', '/usr/sbin', $'($env.HOME)/go/bin', $'($env.GOBIN)', $'($env.HOME)/.krew/bin'])
```
and path is correctly set

**Reply by ghost** [reply]
in that case i don't really have an idea.

in the past it could be due to `$env.ENV_CONVERSIONS.PATH` not being set yet, but in a recent update (0.100?) they changed it so that that's set by default before the config even gets loaded..

**Reply by Guara92** [reply]
I had the time to troubleshoot this more, I discovered that configuring `LS_COLORS` with vivid as documented in the [official doc](https://www.nushell.sh/book/coloring_and_theming.html#vivid-themes), actually breaks the config. Commenting out that line fix the problem, I'm missing something? Or should I open a bug?

**Reply by fdncred** [reply]
open a bug because i'm interested in understanding how it breaks things. i've used vivid off and on for years without issue so i'm a bit confused.
